[
  {
    "package": {
      "name": "cryptography",
      "version": "41.0.2",
      "ecosystem": "PyPI"
    },
    "vulnerabilities": [
      {
        "modified": "2023-08-01T22:46:29Z",
        "published": "2023-08-01T22:34:41Z",
        "schema_version": "1.4.0",
        "id": "GHSA-jm77-qphf-c4w8",
        "summary": "pyca/cryptography's wheels include vulnerable OpenSSL",
        "details": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 0.8-41.0.2 are vulnerable to several security issues. More details about the vulnerabilities themselves can be found in https://www.openssl.org/news/secadv/20230731.txt, https://www.openssl.org/news/secadv/20230719.txt, and https://www.openssl.org/news/secadv/20230714.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0.8"
                  },
                  {
                    "fixed": "41.0.3"
                  }
                ]
              }
            ],
            "versions": [
              "0.8",
              "0.8.1",
              "0.8.2",
              "0.9",
              "0.9.1",
              "0.9.2",
              "0.9.3",
              "1.0",
              "1.0.1",
              "1.0.2",
              "1.1",
              "1.1.1",
              "1.1.2",
              "1.2",
              "1.2.1",
              "1.2.2",
              "1.2.3",
              "1.3",
              "1.3.1",
              "1.3.2",
              "1.3.3",
              "1.3.4",
              "1.4",
              "1.5",
              "1.5.1",
              "1.5.2",
              "1.5.3",
              "1.6",
              "1.7",
              "1.7.1",
              "1.7.2",
              "1.8",
              "1.8.1",
              "1.8.2",
              "1.9",
              "2.0",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.1",
              "2.1.1",
              "2.1.2",
              "2.1.3",
              "2.1.4",
              "2.2",
              "2.2.1",
              "2.2.2",
              "2.3",
              "2.3.1",
              "2.4",
              "2.4.1",
              "2.4.2",
              "2.5",
              "2.6",
              "2.6.1",
              "2.7",
              "2.8",
              "2.9",
              "2.9.1",
              "2.9.2",
              "3.0",
              "3.1",
              "3.1.1",
              "3.2",
              "3.2.1",
              "3.3",
              "3.3.1",
              "3.3.2",
              "3.4",
              "3.4.1",
              "3.4.2",
              "3.4.3",
              "3.4.4",
              "3.4.5",
              "3.4.6",
              "3.4.7",
              "3.4.8",
              "35.0.0",
              "36.0.0",
              "36.0.1",
              "36.0.2",
              "37.0.0",
              "37.0.1",
              "37.0.2",
              "37.0.3",
              "37.0.4",
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/08/GHSA-jm77-qphf-c4w8/GHSA-jm77-qphf-c4w8.json"
            }
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-jm77-qphf-c4w8"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/b22271cf3c3dd8dc8978f8f4b00b5c7060b6538d"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/bfa4d95f0f356f2d535efd5c775e0fb3efe90ef2"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          },
          {
            "type": "WEB",
            "url": "https://www.openssl.org/news/secadv/20230714.txt"
          },
          {
            "type": "WEB",
            "url": "https://www.openssl.org/news/secadv/20230719.txt"
          },
          {
            "type": "WEB",
            "url": "https://www.openssl.org/news/secadv/20230731.txt"
          }
        ],
        "database_specific": {
          "cwe_ids": [],
          "github_reviewed": true,
          "github_reviewed_at": "2023-08-01T22:34:41Z",
          "nvd_published_at": null,
          "severity": "LOW"
        }
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-jm77-qphf-c4w8"
        ]
      }
    ]
  },
  {
    "package": {
      "name": "kerberos",
      "version": "1.3.1",
      "ecosystem": "PyPI"
    },
    "vulnerabilities": [
      {
        "modified": "2021-07-25T23:34:38Z",
        "published": "2017-08-25T18:29:00Z",
        "schema_version": "1.4.0",
        "id": "PYSEC-2017-49",
        "aliases": [
          "CVE-2015-3206"
        ],
        "details": "The checkPassword function in python-kerberos does not authenticate the KDC it attempts to communicate with, which allows remote attackers to cause a denial of service (bad response), or have other unspecified impact by performing a man-in-the-middle attack.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "kerberos",
              "purl": "pkg:pypi/kerberos"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0.0"
                  }
                ]
              }
            ],
            "versions": [
              "1.1.1",
              "1.1.2",
              "1.2.0",
              "1.2.2",
              "1.2.3",
              "1.2.4",
              "1.2.5",
              "1.3.0",
              "1.3.1"
            ],
            "database_specific": {
              "source": "https://github.com/pypa/advisory-database/blob/main/vulns/kerberos/PYSEC-2017-49.yaml"
            }
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://pypi.python.org/pypi/kerberos"
          },
          {
            "type": "REPORT",
            "url": "https://github.com/apple/ccs-pykerberos/issues/31"
          },
          {
            "type": "REPORT",
            "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1223802"
          },
          {
            "type": "WEB",
            "url": "http://www.securityfocus.com/bid/74760"
          },
          {
            "type": "WEB",
            "url": "http://www.openwall.com/lists/oss-security/2015/05/21/3"
          }
        ]
      }
    ],
    "groups": [
      {
        "ids": [
          "PYSEC-2017-49"
        ]
      }
    ]
  }
]